### DISCUSSION TITLE
Automated reports in local using docker compose - Failed taking a screenshot Message: 'geckodriver' executable needs to be in PATH.

### DISCUSSION THREAD

**Discussion by SekelPrem** [apache/superset#24932] [General]
Following the docker-compose for non dev, configured it locally on Mac OSX, I was unable to get the reports to my email even after setting up the SMTP setting

#### How to reproduce the bug

1. Go to 'https://superset.apache.org/docs/installation/installing-superset-using-docker-compose'
2. Follow the setup to install and run the Apache super set using the docker
3. Enabling Alerts and Reports features with Redis, celery_best schedule, and other configurations as specified in the setup
4. After setting up reports for a sample or custom-created chart I see the below errors.

### Expected results
1. email to be recieved
what you expected to happen.

### Actual results
Errors

what actually happens.

#### Screenshots

Since I was receiving gecko-driver errors, I tried to install them in requirement.txt but it din't work, please let me know what to be added in requirements txt so that gecko-driver is installed, 

<img width="1440" alt="image" src="https://github.com/apache/superset/assets/119047771/d6706c67-6f1e-4475-a660-dd472f237153">
<img width="1433" alt="image" src="https://github.com/apache/superset/assets/119047771/e270bb54-f30d-4009-b57e-4592d81105e3">

### Environment

(please complete the following information):

- browser type and version: chrome
- superset version: `superset version` 2.1.0
- python version: `python --version` 3.8
- node.js version: `node -v` docker node
- any feature flags active: Alerts and Reports

### Checklist

Make sure to follow these steps before submitting your issue - thank you!

- [ Yes ] I have checked the superset logs for python stacktraces and included it here as text if there are any.
- [ Yes ] I have reproduced the issue with at least the latest released version of superset.
- [ Yes ] I have checked the issue tracker for the same issue and I haven't found one similar.

**Comment by sfirke** [comment]
Have you tried running the `-dev` image of whatever Superset version you want to deploy?

From the Alerts & Reports documentation:

> Note: All the components required (Firefox headless browser, Redis, Postgres db, celery worker and celery beat) are present in the dev docker image if you are following [Installing Superset Locally](https://superset.apache.org/docs/installation/installing-superset-using-docker-compose/). All you need to do is add the required config variables described in this guide (See Detailed Config).
>
> If you are running a non-dev docker image, e.g., a stable release like apache/superset:2.0.1, that image does not include a headless browser. Only the superset_worker container needs this headless browser to browse to the target chart or dashboard. You can either install and configure the headless browser - see "Custom Dockerfile" section below - or when deploying via docker-compose, modify your docker-compose.yml file to use a dev image for the worker container and a stable release image for the superset_app container.
>
> Note: In this context, a "dev image" is the same application software as its corresponding non-dev image, just bundled with additional tools. So an image like 2.0.1-dev is identical to 2.0.1 when it comes to stability, functionality, and running in production. The actual "in-development" versions of Superset - cutting-edge and unstable - are not tagged with version numbers on Docker Hub and will display version 0.0.0-dev within the Superset UI.

This is not a bug report, it's a configuration question, so I'm going to close this and convert to discussion.  Feel free to keep talking in the discussion if you have further issues.
